### PR TITLE
Document salttesting helpers

### PR DESCRIPTION
**Pull request by cachedout**

### CODE REVIEW

File: doc/topics/development/tests/index.rst [hunk 1]
```diff
@@ -415,3 +415,40 @@ question if the path forward is unclear.
...
     testing once the test file has propagated to newer release branches.
+
+
+Test Helpers
+------------
+
+Several Salt-specific helpers are available. A full list is available by inspecting
+functions exported in `salttesting.helpers`.
+
+`@expensiveTest` -- Designates a test which typically requires a relatively costly
```
Comment (rallytime) [inline doc/topics/development/tests/index.rst]: Either remove the `a` on this line, or change `resources` to `resource` in the next line.

File: doc/topics/development/tests/index.rst [hunk 2]
```diff
@@ -415,3 +415,40 @@ question if the path forward is unclear.
...
+
+`@requires_system_grains` -- Loads and passes the grains on the system as an
+keyword argument to the test function with the name `grains`. 
+    
+`@skip_if_binaries_missing(['list', 'of', 'binaries'])` -- If called from inside a test,
+the test will be skipped if the binaries are not all present on the system.
+
+`@skip_if_not_root` -- If the test is not executed as root, it will be skipped.
+
+`@with_system_user` -- Creates an optionally destroy a system user within a test case.
```
Comment (rallytime) [inline doc/topics/development/tests/index.rst]: `Creates an optionally destroy` --> `Creates an option to destroy`?

File: doc/topics/development/tests/index.rst [hunk 3]
```diff
@@ -415,3 +415,40 @@ question if the path forward is unclear.
...
+See implementation details in `salttesting.helpers` for details.
+
+`@with_system_group` -- Creates an optionally destroy a system group within a test case.
```
Comment (rallytime) [inline doc/topics/development/tests/index.rst]: Same comment here as above.

File: doc/topics/development/tests/index.rst [hunk 4]
```diff
@@ -415,3 +415,40 @@ question if the path forward is unclear.
...
+See implementation details in `salttesting.helpers` for details.
+
+`@with_system_user_and_group` -- Creates an optionally destroy a system user and group
```
Comment (rallytime) [inline doc/topics/development/tests/index.rst]: And here, too. :)

**Review by rallytime — COMMENTED** [review]
I left some comments for a couple of sentences that could be cleaned up. Otherwise, these docs look great and will be nice to have. :D

**Comment by cachedout** [comment]
Issues fixed. Thanks, @rallytime

**Pull request merged**